src/app.rs

```diff
@@ -1580,10 +1604,44 @@ impl App {
     pub fn is_help(&self) -> bool {
         self.display_help
     }
-}
 
-pub fn name_by_id(names: &HashMap<Uuid, String>, id: Uuid) -> &str {
-    names.get(&id).map(|s| s.as_ref()).unwrap_or("Unknown Name")
+    pub(crate) async fn request_contacts_sync(&mut self) -> anyhow::Result<()> {
+        let now = Utc::now();
+        let do_sync = self
+            .data
+            .contacts_sync_request_at
+            .map(|dt| dt + Duration::seconds(CONTACTS_SYNC_DEADLINE_SEC) < now)
+            .unwrap_or(true);
+        if do_sync {
+            info!("requesting contact sync");
+            self.signal_manager.request_contacts_sync().await?;
+            self.data.contacts_sync_request_at = Some(now);
+            self.save().unwrap();
+        }
+        Ok(())
+    }
+
+    /// Filters visible channel based on the provided `pattern`
+    ///
+    /// `pattern` is compared to channel name or channel member contact names, case insensitively.
+    pub(crate) fn filter_channels(&mut self, pattern: &str) {
+        let pattern = pattern.to_lowercase();
+
+        // move out `channels` temporarily to make borrow checker happy
+        let mut channels = std::mem::take(&mut self.data.channels);
+        channels.filter(|channel: &Channel| match pattern.chars().next() {
+            None => true,
+            Some('@') => match channel.group_data.as_ref() {
+                Some(group_data) => group_data
+                    .members
+                    .iter()
+                    .any(|&id| self.name_by_id(id).to_lowercase().contains(&pattern[1..])),
+                None => channel.name.to_lowercase().contains(&pattern[1..]),
+            },
+            _ => channel.name.to_lowercase().contains(&pattern),
+        });
+        self.data.channels = channels;
+    }
```

Comment:
Moving this part here is a nice idea. Have you any other idea or special search pattern that could be implemented? I thought of searching messages, this would need porting the filtering abstraction to messages.

Reply:
Maybe we can take a look at a command palette in popular editors. They have often search prefixes like `@name` etc...